### PR TITLE
Possible fix for SI-6344, value class generic signatures.

### PR DESCRIPTION
**Pull request by paulp**
I'm not enormously confident that I'm generating the
right signatures, but it's closer at worst.

### CODE REVIEW

**Comment by paulp** [comment]
Review by @retronym et al

File: test/files/run/t6344.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+
+class Val[A](val value: Int) extends AnyVal
+class C2 {
+  def foo[A](in: Val[A]) = in
+  def bar[A]: List[Val[A]] = Nil
+  def baz[A]: Val[A] = new Val[A](0)
+  def f1[T <: String](x: Val[T], ys: List[Val[T]]) = ys.head
```
Comment (retronym) [inline test/files/run/t6344.scala]: Could you add some tests for a value class wrapping `String` and one wrapping `Any`? I'm not sure these are treated correctly.

**Comment by paulp** [comment]
PLS REBUILD ALL

**Comment by retronym** [comment]
>  Not everything in the checkfile looks correct

Can you add your analysis of the still-wrong parts to the Jira ticket and the test case? `C5` looks suspicious to me, but perhaps you've spotted more problems.

Is it possible to run the signature verifier as part of the test?

**Comment by odersky** [comment]
LGTM

**Comment by gkossakowski** [comment]
Cool. I agree it's definitively an improvements.

**Pull request merged**